documentation/release_notes/v3.6.0

```diff
@@ -0,0 +1,11 @@
+# Release Notes v3.6.0
+
+
+
+### Critical changes
+ - `hlp_(de)serialize` and `hlp_serial_size` rewritten and now called via `(de)serialise` and `serial_size` wrappers which call relevant MEX/non-MEX functions based on config.
```

Comment:
It would be good to address the main purpose of this change, i.e. to improve the performance of object serialization and the performance of MPI jobs. I think this is what is most relevant to users - especially because I don't see a use-case for users using this serializer directly

Reply:
Good point. Will update.